### PR TITLE
rudimentary matrix export as text facilities

### PR DESCRIPTION
**Pull request by danking**
cc: @jbloom22

### CODE REVIEW

File: python/hail/ldMatrix.py [hunk 1]
```diff
@@ -83,3 +84,110 @@ def read(path):
...
+        Write the upper-triangle with the diagonal as a comma-separated file:
+
+        >>> ldm = vds.ld_matrix()
+        >>> ldm.export('output/ld_matrix.tsv',
+        ...            column_delimiter=',',
+        ...            entries='strict_upper')
+
+        **Notes**
+
+        A full, 3x3 LD matrix written as a comma-separated file looks like this:
```
Comment (jigold) [inline python/hail/ldMatrix.py]: Should you put something here saying the max number of columns that can be exported is INT_MAX?

File: python/hail/ldMatrix.py [hunk 2]
```diff
@@ -83,3 +84,110 @@ def read(path):
...
+        Write a full LD matrix as a tab-separated file:
+
+        >>> vds.ld_matrix().export('output/ld_matrix.tsv', column_delimiter='\t')
+
+        Write a full LD matrix as a comma-separated file with the variant list as a header:
+
+        >>> ldm = vds.ld_matrix()
+        >>> ldm.export('output/ld_matrix.tsv',
+        ...            column_delimiter=',',
+        ...            header=','.join(ldm.variant_list))
```
Comment (jigold) [inline python/hail/ldMatrix.py]: You're missing the parentheses after `variant_list`. That should fix your tests.

File: src/test/scala/is/hail/methods/LDMatrixSuite.scala [hunk 3]
```diff
@@ -96,4 +98,130 @@ class LDMatrixSuite extends SparkSuite {
...
+      Array(5.0, 6.0),
+      Array(9.0))
+
+    exportImportAssert(ldm.exportUpperTriangle(_, ",", header=None, parallelWrite=false),
+      Array(2.0, 3.0),
+      Array(6.0))
+  }
+
+  @Test
+  def exportAllZeros() {
```
Comment (jigold) [inline src/test/scala/is/hail/methods/LDMatrixSuite.scala]: What is the benefit of the zeros test? It seems like your test above is sufficient.

Comment (danking) [reply src/test/scala/is/hail/methods/LDMatrixSuite.scala]: Heh, this is a classic instance of [white-box testing](https://en.wikipedia.org/wiki/White-box_testing). I know that the implementation of the `LDMatrix` uses `IndexedRowMatrix` which doesn't even store all-zero rows. So I wanted to make sure all-zero rows are propagated in all circumstances.

**Pull request merged**